frontend/src/Editor/Components/RadioButton.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { resolveReferences } from '@/_helpers/utils';
+
+
+export const RadioButton = function RadioButton({
+  id,
+  width,
+  height,
+  component,
+  onComponentClick,
+  currentState,
+  onComponentOptionChanged,
+  onEvent
+}) {
+
+  const label = component.definition.properties.label.value;
+  const textColorProperty = component.definition.styles.textColor;
+  const textColor = textColorProperty ? textColorProperty.value : '#000';
+
+  const values = component.definition.properties.values.value;
+  const displayValues = component.definition.properties.display_values.value;
+
+  let parsedValues = values;
+
+  try {
+    parsedValues = resolveReferences(values, currentState, []);
+  } catch (err) { console.log(err); }
+
+  let parsedDisplayValues = displayValues;
+
+  try {
+    parsedDisplayValues = resolveReferences(displayValues, currentState, []);
+  } catch (err) { console.log(err); }
+
+  let selectOptions = [];
+
+  try {
+    selectOptions = [
+      ...parsedValues.map((value, index) => {
+        return { name: parsedDisplayValues[index], value: value };
+      })
+    ];
+  } catch (err) { console.log(err); }
+
+
+  function onSelect(event) {
+    const checked = event.target.value
+    onComponentOptionChanged(component, 'value', checked);
+    if (checked) {
+      onEvent('onCheck', { component });
```

Comment:
`onCheck` event works better for checkboxes, here the event should be `onSelectionChange`.

Reply:
okay 👍